src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTagger.cs

```diff
@@ -19,56 +23,104 @@ internal sealed class InlineParameterNameHintsTagger : ITagger<IntraTextAdornmen
     {
         private readonly ITagAggregator<InlineParameterNameHintDataTag> _tagAggregator;
         private readonly ITextBuffer _buffer;
+        private readonly ITextView _textView;
+
+        /// <summary>
+        /// _cache stores the parameter hint tags in a global location 
+        /// </summary>
+        private readonly List<ITagSpan<IntraTextAdornmentTag>> _cache;
+
+        /// <summary>
+        /// Stores the snapshot of the state at which things change on the screen 
+        /// </summary>
+        private ITextSnapshot? _cacheSnapshot;
+
+        private readonly IClassificationFormatMap _formatMap;
+        private TextFormattingRunProperties? _format;
+        private readonly IClassificationType _hint;
 
-        public event EventHandler<SnapshotSpanEventArgs> TagsChanged;
+        public event EventHandler<SnapshotSpanEventArgs>? TagsChanged;
 
-        public InlineParameterNameHintsTagger(ITextBuffer buffer, ITagAggregator<InlineParameterNameHintDataTag> tagAggregator)
+        public InlineParameterNameHintsTagger(InlineParameterNameHintsTaggerProvider taggerProvider, ITextView textView, ITextBuffer buffer, ITagAggregator<InlineParameterNameHintDataTag> tagAggregator)
         {
+            _cache = new List<ITagSpan<IntraTextAdornmentTag>>();
+            _textView = textView;
             _buffer = buffer;
             _tagAggregator = tagAggregator;
+            _formatMap = taggerProvider.ClassificationFormatMapService.GetClassificationFormatMap(textView);
+            _hint = taggerProvider.ClassificationTypeRegistryService.GetClassificationType(InlineParameterNameHintsTag.TagId);
+            _formatMap.ClassificationFormatMappingChanged += this.OnClassificationFormatMappingChanged;
             _tagAggregator.TagsChanged += OnTagAggregatorTagsChanged;
         }
 
+        private void OnClassificationFormatMappingChanged(object sender, EventArgs e)
+        {
+            if (_format != null)
+            {
+                _format = null;
+                _cache.Clear();
+            }
+        }
+
         private void OnTagAggregatorTagsChanged(object sender, TagsChangedEventArgs e)
         {
+            _cache.Clear();
             var spans = e.Span.GetSpans(_buffer);
             foreach (var span in spans)
             {
                 TagsChanged?.Invoke(this, new SnapshotSpanEventArgs(span));
             }
         }
 
+        private TextFormattingRunProperties Format
+        {
+            get
+            {
+                _format ??= _formatMap.GetTextProperties(_hint);
+
+                return _format;
+            }
+        }
```

Comment:
```suggestion
        private TextFormattingRunProperties Format => _format ??= _formatMap.GetTextProperties(_hint);
```

However, i would also want a threading assertion here.